pkg/lexer/lexer.go

```diff
@@ -20,38 +20,46 @@ import (
 	"github.com/raklaptudirm/mash/pkg/token"
 )
 
+// Various error values returned by lexer.consume.
 var (
 	ErrIllegalNUL = fmt.Errorf("illegal character NUL")
 	ErrIllegalBOM = fmt.Errorf("illegal byte order mark")
 	ErrIllegalEnc = fmt.Errorf("illegal utf-8 encoding")
```

Comment:
Change these to `errors.New` instead `fmt` is used for error values with additional context whereas these are static error values. Also, have comments for each - the gopls makes it easy to get the meaning behind these through the doc comments.

Last, could you instead remove the `Illegal` in the name of these errors? I think it is enough to see `Err` in the name to tell us that its unexpected - `Illegal` doesn't add any new context to these errors...

Reply:
Changed `fmt.Errorf`s to `errors.New`.

Reply:
I think you misunderstood me. I was taking about the variable name not the message inside the error.

`ErrIllegalNUL` should be `ErrNUL` in my opinion.

Reply:
Ah yes, we could change that.